### PR TITLE
Issue #13213: Resolve the '//ok' comments from annotationlocation pac…

### PR DESCRIPTION
**Pull request by Rohanraj123**
part of #13213 

Link to check the documentation :  [clickHere](https://checkstyle.org/checks/annotation/annotationlocation.html#AnnotationLocation)

proof that all supressions for this check are removed 

```
rajro@DESKTOP-HSV42DL MINGW64 ~/checkstyle (master)
$ grep annotationlocation config/checkstyle-input-suppressions.xml
            files="checks[\\/]annotation[\\/]annotationlocation[\\/]InputAnnotationLocationRecordsAndCompactCtors.java"/>
            files="checks[\\/]annotation[\\/]annotationlocation[\\/]InputAnnotationLocationCorrect.java"/>
            files="checks[\\/]annotation[\\/]annotationlocation[\\/]InputAnnotationLocationCorrect.java"/>
            files="checks[\\/]annotation[\\/]annotationlocation[\\/]InputAnnotationLocationCorrect.java"/>
            files="checks[\\/]annotation[\\/]annotationlocation[\\/]InputAnnotationLocationCorrect.java"/>
            files="checks[\\/]annotation[\\/]annotationlocation[\\/]InputAnnotationLocationCorrect.java"/>
            files="checks[\\/]annotation[\\/]annotationlocation[\\/]InputAnnotationLocationCorrect.java"/>
            files="checks[\\/]annotation[\\/]annotationlocation[\\/]InputAnnotationLocationCorrect.java"/>
            files="checks[\\/]annotation[\\/]annotationlocation[\\/]InputAnnotationLocationCorrect.java"/>
            files="checks[\\/]annotation[\\/]annotationlocation[\\/]InputAnnotationLocationCorrect.java"/>
            files="checks[\\/]annotation[\\/]annotationlocation[\\/]InputAnnotationLocationCorrect.java"/>
            files="checks[\\/]annotation[\\/]annotationlocation[\\/]InputAnnotationLocationCorrect.java"/>
            files="checks[\\/]annotation[\\/]annotationlocation[\\/]InputAnnotationLocationCorrect.java"/>
            files="checks[\\/]annotation[\\/]annotationlocation[\\/]InputAnnotationLocationCorrect.java"/>
            files="checks[\\/]annotation[\\/]annotationlocation[\\/]InputAnnotationLocationCorrect.java"/>
            files="checks[\\/]annotation[\\/]annotationlocation[\\/]InputAnnotationLocationCorrect.java"/>
            files="checks[\\/]annotation[\\/]annotationlocation[\\/]InputAnnotationLocationCorrect.java"/>
            files="checks[\\/]annotation[\\/]annotationlocation[\\/]InputAnnotationLocationCorrect.java"/>
            files="checks[\\/]annotation[\\/]annotationlocation[\\/]InputAnnotationLocationCorrect.java"/>
            files="checks[\\/]annotation[\\/]annotationlocation[\\/]InputAnnotationLocationCorrect.java"/>
            files="checks[\\/]annotation[\\/]annotationlocation[\\/]InputAnnotationLocationCorrect.java"/>
            files="checks[\\/]annotation[\\/]annotationlocation[\\/]InputAnnotationLocationCorrect.java"/>
            files="checks[\\/]annotation[\\/]annotationlocation[\\/]InputAnnotationLocationCorrect.java"/>
            files="checks[\\/]annotation[\\/]annotationlocation[\\/]InputAnnotationLocationCorrect.java"/>
            files="checks[\\/]annotation[\\/]annotationlocation[\\/]InputAnnotationLocationCorrect.java"/>
            files="checks[\\/]annotation[\\/]annotationlocation[\\/]InputAnnotationLocationCorrect.java"/>
            files="checks[\\/]annotation[\\/]annotationlocation[\\/]InputAnnotationLocationCorrect.java"/>
            files="checks[\\/]annotation[\\/]annotationlocation[\\/]InputAnnotationLocationCorrect.java"/>
            files="checks[\\/]annotation[\\/]annotationlocation[\\/]InputAnnotationLocationCorrect.java"/>
            files="checks[\\/]annotation[\\/]annotationlocation[\\/]InputAnnotationLocationCorrect.java"/>
            files="checks[\\/]annotation[\\/]annotationlocation[\\/]InputAnnotationLocationCorrect.java"/>
            files="checks[\\/]annotation[\\/]annotationlocation[\\/]InputAnnotationLocationCorrect.java"/>
            files="checks[\\/]annotation[\\/]annotationlocation[\\/]InputAnnotationLocationCorrect.java"/>
            files="checks[\\/]annotation[\\/]annotationlocation[\\/]InputAnnotationLocationCorrect.java"/>
            files="checks[\\/]annotation[\\/]annotationlocation[\\/]InputAnnotationLocationCustomAnnotationsDeclared.java"/>
            files="checks[\\/]annotation[\\/]annotationlocation[\\/]InputAnnotationLocationCustomAnnotationsDeclared.java"/>
            files="checks[\\/]annotation[\\/]annotationlocation[\\/]InputAnnotationLocationCustomAnnotationsDeclared.java"/>
            files="checks[\\/]annotation[\\/]annotationlocation[\\/]InputAnnotationLocationDeprecatedAndCustom.java"/>
            files="checks[\\/]annotation[\\/]annotationlocation[\\/]InputAnnotationLocationDeprecatedAndCustom.java"/>
            files="checks[\\/]annotation[\\/]annotationlocation[\\/]InputAnnotationLocationDeprecatedAndCustom.java"/>
            files="checks[\\/]annotation[\\/]annotationlocation[\\/]InputAnnotationLocationEmpty.java"/>
            files="checks[\\/]annotation[\\/]annotationlocation[\\/]InputAnnotationLocationIncorrect.java"/>
            files="checks[\\/]annotation[\\/]annotationlocation[\\/]InputAnnotationLocationIncorrect.java"/>
            files="checks[\\/]annotation[\\/]annotationlocation[\\/]InputAnnotationLocationIncorrect.java"/>
            files="checks[\\/]annotation[\\/]annotationlocation[\\/]InputAnnotationLocationIncorrect.java"/>
            files="checks[\\/]annotation[\\/]annotationlocation[\\/]InputAnnotationLocationIncorrect.java"/>
            files="checks[\\/]annotation[\\/]annotationlocation[\\/]InputAnnotationLocationIncorrect.java"/>
            files="checks[\\/]annotation[\\/]annotationlocation[\\/]InputAnnotationLocationIncorrect.java"/>
            files="checks[\\/]annotation[\\/]annotationlocation[\\/]InputAnnotationLocationIncorrect.java"/>
            files="checks[\\/]annotation[\\/]annotationlocation[\\/]InputAnnotationLocationIncorrect.java"/>
            files="checks[\\/]annotation[\\/]annotationlocation[\\/]InputAnnotationLocationIncorrect.java"/>
            files="checks[\\/]annotation[\\/]annotationlocation[\\/]InputAnnotationLocationIncorrect.java"/>
            files="checks[\\/]annotation[\\/]annotationlocation[\\/]InputAnnotationLocationIncorrect.java"/>
            files="checks[\\/]annotation[\\/]annotationlocation[\\/]InputAnnotationLocationIncorrect.java"/>
            files="checks[\\/]annotation[\\/]annotationlocation[\\/]InputAnnotationLocationIncorrect.java"/>
            files="checks[\\/]annotation[\\/]annotationlocation[\\/]InputAnnotationLocationIncorrect2.java"/>
            files="checks[\\/]annotation[\\/]annotationlocation[\\/]InputAnnotationLocationIncorrect2.java"/>
            files="checks[\\/]annotation[\\/]annotationlocation[\\/]InputAnnotationLocationIncorrect2.java"/>
            files="checks[\\/]annotation[\\/]annotationlocation[\\/]InputAnnotationLocationIncorrect2.java"/>
            files="checks[\\/]annotation[\\/]annotationlocation[\\/]InputAnnotationLocationIncorrect2.java"/>
            files="checks[\\/]annotation[\\/]annotationlocation[\\/]InputAnnotationLocationIncorrect2.java"/>
            files="checks[\\/]annotation[\\/]annotationlocation[\\/]InputAnnotationLocationIncorrect2.java"/>
            files="checks[\\/]annotation[\\/]annotationlocation[\\/]InputAnnotationLocationIncorrect2.java"/>
            files="checks[\\/]annotation[\\/]annotationlocation[\\/]InputAnnotationLocationIncorrect2.java"/>
            files="checks[\\/]annotation[\\/]annotationlocation[\\/]InputAnnotationLocationIncorrect2.java"/>
            files="checks[\\/]annotation[\\/]annotationlocation[\\/]InputAnnotationLocationIncorrect2.java"/>
            files="checks[\\/]annotation[\\/]annotationlocation[\\/]InputAnnotationLocationIncorrect2.java"/>
            files="checks[\\/]annotation[\\/]annotationlocation[\\/]InputAnnotationLocationIncorrect2.java"/>
            files="checks[\\/]annotation[\\/]annotationlocation[\\/]InputAnnotationLocationIncorrect2.java"/>
            files="checks[\\/]annotation[\\/]annotationlocation[\\/]InputAnnotationLocationIncorrect2.java"/>
            files="checks[\\/]annotation[\\/]annotationlocation[\\/]InputAnnotationLocationIncorrect2.java"/>
            files="checks[\\/]annotation[\\/]annotationlocation[\\/]InputAnnotationLocationIncorrect2.java"/>
            files="checks[\\/]annotation[\\/]annotationlocation[\\/]InputAnnotationLocationIncorrect2.java"/>
            files="checks[\\/]annotation[\\/]annotationlocation[\\/]InputAnnotationLocationIncorrect2.java"/>
            files="checks[\\/]annotation[\\/]annotationlocation[\\/]InputAnnotationLocationIncorrect2.java"/>
            files="checks[\\/]annotation[\\/]annotationlocation[\\/]InputAnnotationLocationIncorrect2.java"/>
            files="checks[\\/]annotation[\\/]annotationlocation[\\/]InputAnnotationLocationIncorrect2.java"/>
            files="checks[\\/]annotation[\\/]annotationlocation[\\/]InputAnnotationLocationIncorrect2.java"/>
            files="checks[\\/]annotation[\\/]annotationlocation[\\/]InputAnnotationLocationIncorrect2.java"/>
            files="checks[\\/]annotation[\\/]annotationlocation[\\/]InputAnnotationLocationIncorrect2.java"/>
            files="checks[\\/]annotation[\\/]annotationlocation[\\/]InputAnnotationLocationIncorrect2.java"/>
            files="checks[\\/]annotation[\\/]annotationlocation[\\/]InputAnnotationLocationIncorrect2.java"/>
            files="checks[\\/]annotation[\\/]annotationlocation[\\/]InputAnnotationLocationIncorrect2.java"/>
            files="checks[\\/]annotation[\\/]annotationlocation[\\/]InputAnnotationLocationIncorrect2.java"/>
            files="checks[\\/]annotation[\\/]annotationlocation[\\/]InputAnnotationLocationIncorrect3.java"/>
            files="checks[\\/]annotation[\\/]annotationlocation[\\/]InputAnnotationLocationIncorrect3.java"/>
            files="checks[\\/]annotation[\\/]annotationlocation[\\/]InputAnnotationLocationIncorrect3.java"/>
            files="checks[\\/]annotation[\\/]annotationlocation[\\/]InputAnnotationLocationIncorrect3.java"/>
            files="checks[\\/]annotation[\\/]annotationlocation[\\/]InputAnnotationLocationIncorrect3.java"/>
            files="checks[\\/]annotation[\\/]annotationlocation[\\/]InputAnnotationLocationIncorrect3.java"/>
            files="checks[\\/]annotation[\\/]annotationlocation[\\/]InputAnnotationLocationIncorrect3.java"/>
            files="checks[\\/]annotation[\\/]annotationlocation[\\/]InputAnnotationLocationIncorrect3.java"/>
            files="checks[\\/]annotation[\\/]annotationlocation[\\/]InputAnnotationLocationIncorrect3.java"/>
            files="checks[\\/]annotation[\\/]annotationlocation[\\/]InputAnnotationLocationIncorrect3.java"/>
            files="checks[\\/]annotation[\\/]annotationlocation[\\/]InputAnnotationLocationIncorrect3.java"/>
            files="checks[\\/]annotation[\\/]annotationlocation[\\/]InputAnnotationLocationIncorrect3.java"/>
            files="checks[\\/]annotation[\\/]annotationlocation[\\/]InputAnnotationLocationIncorrect3.java"/>
            files="checks[\\/]annotation[\\/]annotationlocation[\\/]InputAnnotationLocationIncorrect3.java"/>
            files="checks[\\/]annotation[\\/]annotationlocation[\\/]InputAnnotationLocationIncorrect3.java"/>
            files="checks[\\/]annotation[\\/]annotationlocation[\\/]InputAnnotationLocationIncorrect3.java"/>
            files="checks[\\/]annotation[\\/]annotationlocation[\\/]InputAnnotationLocationMultiple.java"/>
            files="checks[\\/]annotation[\\/]annotationlocation[\\/]InputAnnotationLocationSingleParameterless.java"/>
            files="checks[\\/]annotation[\\/]annotationlocation[\\/]InputAnnotationLocationWithoutAnnotations.java"/>
            files="checks[\\/]annotation[\\/]annotationlocation[\\/]InputAnnotationLocationWithoutAnnotations.java"/>
            files="checks[\\/]annotation[\\/]annotationlocation[\\/]inputs[\\/]PackageAnnotations.java"/>

rajro@DESKTOP-HSV42DL MINGW64 ~/checkstyle (master)
$ git checkout issue-13213-annotationlocation
Switched to branch 'issue-13213-annotationlocation'

rajro@DESKTOP-HSV42DL MINGW64 ~/checkstyle (issue-13213-annotationlocation)
$ grep annotationlocation config/checkstyle-input-suppressions.xml

rajro@DESKTOP-HSV42DL MINGW64 ~/checkstyle (issue-13213-annotationlocation)
$ echo $?
1


```

### CODE REVIEW

**Review by nrmancuso — APPROVED** [review]
Ok to merge as CI passes.

@Rohanraj123 please do not open and close PRs.

**Comment by nrmancuso** [comment]
@romani we can use this PR as example in issue

**Comment by Rohanraj123** [comment]
> Ok to merge as CI passes.
> 
> @Rohanraj123 please do not open and close PRs.

Sir sorry but I am newbie at this open source contribution and am getting confused  why these tests fails again and again even after doing everything right. And i closed last PR by mistake. Sorry for the inconvenience. I will not close this PR now. Now suggest me the changes i can do in this PR please.

**Comment by nrmancuso** [comment]
Please read: https://checkstyle.sourceforge.io/beginning_development.html

Make sure that `mvn clean verify` is passing before pushing changes. Take time to read and understand failure messages.

>tests fails again and again even after doing everything right

If tests are failing, everything is not right :)

It looks like you removed a suppression for a file that has not been updated.

File: config/checkstyle-input-suppressions.xml [hunk 1]
```diff
@@ -139,6 +139,8 @@
           files="[\\/]meta[\\/]javadocmetadatascraper[\\/]InputJavadocMetadataScraperWriteTagCheck.java"/>
 
   <!-- until https://github.com/checkstyle/checkstyle/issues/13213 -->
+  <suppress id="UnnecessaryOkComment"
+            files="checks[\\/]annotation[\\/]annotationlocation[\\/]InputAnnotationLocationRecordsAndCompactCtors.java"/>
```
Comment (nrmancuso) [inline config/checkstyle-input-suppressions.xml:143]: We need to update this input file, not add suppression back.

**Review by nrmancuso — CHANGES_REQUESTED** [review]
You should have exactly one commit in this PR, please squash into 1.

**Comment by Rohanraj123** [comment]
> You should have exactly one commit in this PR, please squash into 1.

sure am waiting for CI passes

**Comment by nrmancuso** [comment]
> > You should have exactly one commit in this PR, please squash into 1.
> 
> sure am waiting for CI passes

CI will not pass, I see that you are still not running `mvn clean verify` before pushing changes. Please do this and squash commits into 1.

**Comment by Rohanraj123** [comment]
`mvn clean verify`  Successfull

Now I think squashing should solve the issue.

**Comment by Rohanraj123** [comment]
@nrmancuso Sir, all checks are cleared and also did successfull `mvn clean verify`  Can you please review it.

**Review by romani — APPROVED** [review]
ok to merge

**Review by nrmancuso — APPROVED** [review]

**Pull request merged**